src/main/osd/osd_elements.c

```diff
@@ -156,6 +157,40 @@ static uint32_t blinkBits[(OSD_ITEM_COUNT + 31) / 32];
 #define IS_BLINK(item) (blinkBits[(item) / 32] & (1 << ((item) % 32)))
 #define BLINK(item) (IS_BLINK(item) && blinkState)
 
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+
+typedef int (*getEscRpmOrFreqFnPtr)(int i);
+
+int getEscRpm(int i)
+{
+#ifdef USE_RPM_FILTER
+    if (motorConfig()->dev.useDshotTelemetry) {
+        return 100.0f / (motorConfig()->motorPoleCount / 2.0f) * getDshotTelemetry(i);
+    }
+#endif 
+    if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
+        return calcEscRpm(getEscSensorData(i)->rpm);
+    } else { 
+        return 0;
+    }
+}
+
+int getEscRpmFreq(int i) 
+{
+    return getEscRpm(i) / 60;
+}
+
+void renderOsdEscRpmOrFreq(getEscRpmOrFreqFnPtr escFnPtr, osdElementParms_t *element)
+{
+    int x = element->elemPosX;
+    int y = element->elemPosY;
+    for (int i=0; i < getMotorCount(); i++) {
+        char rpmStr[5];
+        tfp_sprintf(rpmStr, "%d", (*escFnPtr)(i));
+        displayWrite(element->osdDisplayPort, x, y + i, rpmStr);
+    }
```

Comment:
Need to add:
```
    element->drawElement = false;  // element already drawn
```
To indicate that this element is handling the screen drawing directly and that the element display logic doesn't try to render the `buff`. This is a special case in that only a handful of elements do the actual screen drawing (usually multi-line elements). There's no problem with how you're doing it, just need to set the flag to indicate it.

Reply:
it is fixed